src/k8s-extension/azext_k8s_extension/partner_extensions/Dapr.py

```diff
@@ -4,43 +4,153 @@
 
+from typing import Dict, Tuple
+
+from azure.cli.core.azclierror import InvalidArgumentValueError
+from knack.log import get_logger
+from knack.prompting import prompt, prompt_y_n
+
+from ..vendored_sdks.models import Extension, Scope, ScopeCluster
 from .DefaultExtension import DefaultExtension
 
-from ..vendored_sdks.models import (
-    Extension,
-)
+logger = get_logger(__name__)
 
 
 class Dapr(DefaultExtension):
     def __init__(self):
+        self.TSG_LINK = "https://docs.microsoft.com/en-us/azure/aks/dapr"
+        self.DEFAULT_RELEASE_NAME = 'dapr'
+        self.DEFAULT_RELEASE_NAMESPACE = 'dapr-system'
+        self.DEFAULT_RELEASE_TRAIN = 'stable'
+        self.DEFAULT_CLUSTER_TYPE = 'managedclusters'
+        self.DEFAULT_HA = 'true'
+
         # constants for configuration settings.
-        self.CLUSTER_TYPE = 'global.clusterType'
+        self.CLUSTER_TYPE_KEY = 'global.clusterType'
+        self.HA_KEY_PREFIX = 'global.ha.'
+        self.PLACEMENT_KEY_PREFIX = 'dapr_placement.'
+        self.SKIP_EXISTING_DAPR_CHECK_KEY = 'skipExistingDaprCheck'
+        self.EXISTING_DAPR_RELEASE_NAME_KEY = 'existingDaprReleaseName'
+        self.EXISTING_DAPR_RELEASE_NAMESPACE_KEY = 'existingDaprReleaseNamespace'
+
+        # constants for message prompts.
+        self.MSG_IS_DAPR_INSTALLED = "Is Dapr already installed in the cluster?"
+        self.MSG_ENTER_RELEASE_NAME = "Enter the Helm release name for Dapr, "\
+            f"or press Enter to use the default name [{self.DEFAULT_RELEASE_NAME}]: "
+        self.MSG_ENTER_RELEASE_NAMESPACE = "Enter the namespace where Dapr is installed, "\
+            f"or press Enter to use the default namespace [{self.DEFAULT_RELEASE_NAMESPACE}]: "
+        self.RELEASE_INFO_HELP_STRING = "The Helm release name and namespace can be found by running 'helm list -A'."
+
+        # constants for error messages.
+        self.ERR_MSG_INVALID_SCOPE_TPL = "Invalid scope '{}'. This extension can be installed only at 'cluster' scope."\
+            f" Check {self.TSG_LINK} for more information."
+
+    def _get_release_info(self, release_name: str, release_namespace: str, configuration_settings: dict)\
+            -> Tuple[str, str, bool]:
+        '''
+        Check if Dapr is already installed in the cluster and get the release name and namespace.
+        If user has provided the release name and namespace in configuration settings, use those.
+        Otherwise, prompt the user for the release name and namespace.
+        If Dapr is not installed, return the default release name and namespace.
+        '''
+        name, namespace, dapr_exists = release_name, release_namespace, False
+
+        # Set the default release name and namespace if not provided.
+        name = name or self.DEFAULT_RELEASE_NAME
+        namespace = namespace or self.DEFAULT_RELEASE_NAMESPACE
+
+        if configuration_settings.get(self.SKIP_EXISTING_DAPR_CHECK_KEY, 'false') == 'true':
```

Comment:
Does this statement make passing `SKIP_EXISTING_DAPR_CHECK_KEY` mandatory even for new installation ?

Reply:
Yup, the problem is that there is no easy way to tell if Dapr exists already. So we need to check each time with the user if Dapr is there in the cluster. This `skipExistingDaprCheck` configuration can prevent this prompt, any non-interactive flow will have to use this configuration since TTYs won't be available.